frame/nfts/src/lib.rs

```diff
@@ -630,8 +647,15 @@ pub mod pallet {
 
 			Self::do_transfer(collection, item, dest, |collection_details, details| {
 				if details.owner != origin && collection_details.admin != origin {
-					let approved = details.approved.take().map_or(false, |i| i == origin);
+					let approved = details.approvals.contains_key(&origin);
 					ensure!(approved, Error::<T, I>::NoPermission);
+
+					let deadline =
+						details.approvals.get(&origin).ok_or(Error::<T, I>::NoPermission)?;
+					if let Some(d) = deadline {
+						let block_number = frame_system::Pallet::<T>::block_number();
+						ensure!(*d >= block_number, Error::<T, I>::ApprovalExpired);
```

Comment:
It reads a bit easier if we switch things round so it reads 'is block_number less than deadline':
```suggestion
						ensure!(block_number <= *d, Error::<T, I>::ApprovalExpired);
```